### ISSUE TITLE
Out of bounds increment on iterator

### ISSUE THREAD

**Issue by zero9178** [hanickadot/compile-time-regular-expressions#97]
I am currently using the single header file of CTRE at commit d71fca4c622a12c2cf85a2361e8fe7ff50a11976

Following simple code causes an iterator to be incremented past the end iterator of the string_view I am using:
```cpp
#include "ctre.hpp"

constexpr static auto pattern = ctll::fixed_string{R"(\{\})"};

#include <iostream>

int main()
{
    const auto* text = "Stray '{}' in program";
    std::string_view sv = text;
    for (auto& iter : ctre::range<pattern>(sv))
    {
        std::cout << iter.view() << std::endl;
    }
}
```
This causes a runtime exception when using the Debug STL of MSVC or the debug containers of libstdc++ with following code:
```cpp
#include "ctre.hpp"

constexpr static auto pattern = ctll::fixed_string{R"(\{\})"};

#include <iostream>
#include <string>
#include <debug/string>

int main()
{
    const auto* text = "Stray '{}' in program";
    __gnu_debug::string s = text;
    for (auto& iter : ctre::range<pattern>(s))
    {
        std::cout << iter.view() << std::endl;
    }
}
```
Using each code samples above I chased it back to this line of code:
https://github.com/hanickadot/compile-time-regular-expressions/blob/d71fca4c622a12c2cf85a2361e8fe7ff50a11976/single-header/ctre.hpp#L3175-L3183
As you can see it increments although end might be equal to current. Hope this helps!

Thank you for your great work

**Comment by hanickadot** [comment]
Thank you very much for very nice bug report, I was able to fix issue, unfortunately I wasn't able to came with a test to trigger compilation failure (as most of my tests are constexpr). Fixed in 509b646b3b16b4ed48b9420acd0e2c06204b2d0f. 

Please confirm it's working for you.

**Comment by zero9178** [comment]
Yes that worked thank you very much!
